tests/nlforces/test_general_poly_stiffness.py

```diff
@@ -0,0 +1,406 @@
+            'Incorrect jacobian evaluation from nl_force.force, analytical check')
+        
+        """
+        Case 2: 
+            -Compare to Numerical gradient
+        """
+        rng = np.random.default_rng()
+        u0 = rng.random((Nd))
+        Fnl_f, dFnldU_f = nl_force.force(u0) #output from function 
+        Emat = rng.random((Nd,1))
+        
+        grad_check = vutils.check_grad(nl_force.force, u0, verbose=False, 
+                                        rtol=100*self.rtol_grad)    
+        
+        self.assertFalse(grad_check, 
+            'Incorrect jacobian evaluation from nl_force.force, numerical check')
+              
+    def test_aftgradient(self):   
+        """
+        Test analytical expansion of nonlinear force gradient for first harmonic motion
+         for the simple case. Also validates numerical gradient for random values of u and Emat
+        
+        Returns
+        -------
+        None.
+        """
+        # Test System
+        
+        # Simple Mapping to spring displacements
+        Q = np.array([[1, 0,0], \
+                      [0, 1,0], \
+                      [0, 0, 1 ]])
+        
+        # Weighted / integrated mapping back for testing purposes
+        T = np.array([[1, 0,0], \
+                      [0, 1,0], \
+                      [0, 0, 1 ]])
+        
+        qq = np.array([[2, 0, 0], \
+                       [0, 2, 0], \
+                       [3, 0, 0], \
+                       [0, 0, 3],  \
+                       [1, 1, 0], \
+                       [0, 1, 1],  \
+                       [2, 1, 0], \
+                       [1, 1,1]])
+            
+        Emat = np.ones((3,qq.shape[0]))
+        Nd = Q.shape[1]
+            
+        nl_force = GenPolyForce(Q, T, Emat, qq)
+        
+        # h = np.array([0, 1, 2, 3]) # Manual Checking expansion / debugging
+        h = np.array([0, 1, 2, 3, 4, 5, 6, 7]) # Automate Checking with this
+        Nhc = 2*(h !=0).sum() + (h==0).sum() # Number of Harmonic Components
+        w = 1 # Test for various w
+        
+        Nd = Q.shape[1]
+       
+        #U[:3] = 1.0
+        rng = np.random.default_rng()
+        U = rng.random((Nd*Nhc, 1))
+        
+        # Numerically Verify Gradient 
+        fun = lambda U: nl_force.aft(U, w, h)[0:2]
+        grad_failed = vutils.check_grad(fun, U, verbose=False, 
+                                        rtol=self.rtol_grad)
+        
+        self.assertFalse(grad_failed, 
+                         'Incorrect gradiant w.r.t. displacement.')
+        
+        # Numerically Verify Frequency Gradient
+        fun = lambda w: nl_force.aft(U, w[0], h)[0::2]
+        grad_failed = vutils.check_grad(fun, np.array([w]), verbose=False, 
+                                        rtol=self.rtol_grad)
+        
+        self.assertFalse(grad_failed, 
+                         'Incorrect displacement w.r.t. frequency.')
+
+    def test_multiharmonic_grads(self):
```

Comment:
What is the difference between this test and the previous one (elaborate in comments about difference if there is one, or remove if the same)?

Comment on 259-260 suggests that it is multiharmonic w/ skipped harmonics, but h on 290 doesn't skip any harmonics.

Reply:
h skips 4th harmonic